slither/solc_parsing/slither_compilation_unit_solc.py

```diff
@@ -432,7 +436,12 @@ def parse_contracts(self) -> None:  # pylint: disable=too-many-statements,too-ma
                     target = contract_parser.underlying_contract.file_scope.get_contract_from_name(
                         contract_name
                     )
-                    assert target
+                    if target == contract_parser.underlying_contract:
+                        raise InheritanceResolutionError(
+                            "Could not resolve contract inheritance. This is likely caused by contract name reuse (see https://github.com/crytic/slither/issues/1758)."
```

Comment:
I think the issue is more related to import rename that lead to confusion, rather that contract name reuse. Maybe we can word it as "This is likely caused by an import renaming that collide with existing names", or something like that
What do you think?